src/drawer.js

```diff
@@ -383,6 +383,7 @@ $.Drawer.prototype = /** @lends OpenSeadragon.Drawer.prototype */{
 
         this.context.save();
         this.context.globalAlpha = opacity;
+        this.context.globalCompositeOperation = compositeOperation;
```

Comment:
@iangilman, I am sorry, I think I committed something bad. can you revert it? thanks.

----- Original Message -----
From: "Ian Gilman" <notifications@github.com>
To: "openseadragon/openseadragon" <openseadragon@noreply.github.com>
Cc: "meihuisu" <mei@isi.edu>
Sent: Monday, January 18, 2016 9:18:32 AM
Subject: Re: [openseadragon] adding setting of globalCompositeOperation for tiledImage (#814)

> @@ -383,6 +383,7 @@ $.Drawer.prototype = /** @lends OpenSeadragon.Drawer.prototype */{
>  
>          this.context.save();
>          this.context.globalAlpha = opacity;
> +        this.context.globalCompositeOperation = compositeOperation;

It needs to be optional to this function, because arguments in front of it are optional. Even though this function is currently only used in one place, and all argument are passed in that one place, we need to think of the possibility someone else might call it and not pass in all the arguments. 

If it's not passed into this function, I would prefer not to set it at all, rather than setting it to a default value. Depending on the canvas implementation, setting it when we don't need to may have a performance impact. Nothing else in OSD changes the globalCompositeOperation, and this function wraps the change in a `context.save` `context.restore` pair, so the change won't ever bleed out past this call, so we can be confident that it'll always be already set to the default when this function starts. 

Does that make sense? 

---
Reply to this email directly or view it on GitHub:
https://github.com/openseadragon/openseadragon/pull/814/files#r50023838